### PR TITLE
Handle `workspace/didChangeWatchedFiles`

### LINKED ISSUE #446: Handle `workspace/didChangeWatchedFiles`
There are some scenarios where some R files are created, changed, or removed outside editor operations which is not sent from client to server via lsp requests.

* Git checkout
* Rcpp/cpp11 generate R wrapper file for exported/registered C++ functions

In these cases, language server is not aware of the changes of the files, and thus will not update the workspace so that symbol definition and etc will not update accordingly until user explicitly open the changed files.

Therefore, we should handle [`workspace/didChangeWatchedFiles`](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#workspace_didChangeWatchedFiles).

### PR DESCRIPTION
**Pull request by renkun-ken**
Closes #446 

This PR implements `workspace/didChangeWatchedFiles` so that in a package workspace whenever files in `R` folders are changed, the created or modified files will be reloaded, and the removed files are dropped. As a result, the workspace is always up-to-date.

Note that the changes in open documents will trigger `workspace/didChangeWatchedFiles` too, but the client will send requests to properly handle them in the first place. Therefore, we explicitly ignore these open documents here.

- [x] Implement `workspace/didChangeWatchedFiles`
- [x] Test cases

### CODE REVIEW

**Comment by renkun-ken** [comment]
@randy3k Any comments?

**Comment by randy3k** [comment]
LGTM. I think at some point we will need to look into supporting non-package projects, eg a shiny or plumber app.

**Review by randy3k — APPROVED** [review]

File: R/handlers-textsync.R [hunk 1]
```diff
@@ -103,6 +106,9 @@ text_document_did_close <- function(self, params) {
         diagnostics_callback(self, uri, NULL, list())
         self$workspace$documents$remove(uri)
         self$workspace$update_loaded_packages()
+    } else if (self$workspace$documents$has(uri)) {
+        doc <- self$workspace$documents$get(uri)
+        doc$did_close()
```
Comment (randy3k) [inline R/handlers-textsync.R:111]: To simplify the logic, could we move these line to line 103?

Comment (randy3k) [reply R/handlers-textsync.R:111]: You merged quick :) Never mind.

Comment (renkun-ken) [reply R/handlers-textsync.R:111]: Done via 354b8d8

**Pull request merged**